File: Adapter/Python/BicycleAdapter.py
# -*- coding: utf-8 -*-
"""
Created on Sat May  7 15:18:19 2022

@author: MahmoudSaeed
"""

import Vehicles

class BicycleAdapter(Vehicles.Vehicles):
    def __init__(self , b):
        self.b = b
    def accelerate(self):
        self.b.pedal()
    
    def pushBreak(self):
        self.b.stop()
        
    def soundHorn(self):
        self.b.ringBell()

File: Java/adapter/src/adapter/BicycleAdapter.java
package adapter;

public class BicycleAdapter implements Vehicles {
    private Bicycle bicycle;
    public BicycleAdapter(Bicycle bicycle){
        this.bicycle = bicycle;

    }

    @Override
    public void accelerate() {
    bicycle.pedal();
    }

    @Override
    public void pushBreak() {
        bicycle.stop();
    }

    @Override
    public void soundHorn() {
        bicycle.ringBell();
    }
}
